pages/tree-view/keyboard-navigation.page.tsx

```diff
@@ -0,0 +1,104 @@
+
+import { cdsItems, Item as KeyboardNavigationItem, nonCdsItems } from './items/keyboard-navigation-items';
+import { textItems } from './items/permutations-items';
+
+export default function TreeViewKeyboardNavigation() {
+  const [pressedItems, setPressedItems] = useState<ReadonlyArray<string>>([]);
+
+  return (
+    <SpaceBetween size="m">
+      <h1>TreeView keyboard navigation</h1>
+
+      <Button>Focus here</Button>
+
+      <Grid gridDefinition={[{ colspan: { m: 7, xs: 12 } }]}>
+        <Container header={<h2>No focusable elements inside tree-item</h2>}>
+          <TreeView
+            ariaLabel="Tree view without focusable elements inside"
+            items={textItems}
+            renderItem={item => item}
+            getItemId={item => item.id}
+            getItemChildren={item => item.children}
+            i18nStrings={{
+              expandButtonLabel: () => 'Expand item',
+              collapseButtonLabel: () => 'Collapse item',
+            }}
+            connectorLines="vertical"
+          />
+        </Container>
+      </Grid>
+
+      <Button>Focus here</Button>
+
+      <Grid gridDefinition={[{ colspan: { m: 7, xs: 12 } }]}>
+        <Container header={<h2>Focusable CDS elements inside tree-item</h2>}>
+          <TreeView<KeyboardNavigationItem>
+            ariaLabel="Tree view with focusable CDS elements inside"
+            items={cdsItems}
+            renderItem={item => ({
+              icon: item.hasToggleButton ? (
+                <ToggleButton
+                  onChange={({ detail }: any) => {
+                    if (detail.pressed) {
+                      setPressedItems(prev => [...prev, item.id]);
+                    } else {
+                      setPressedItems(prev => prev.filter(id => id !== item.id));
+                    }
+                  }}
+                  pressed={pressedItems.includes(item.id)}
+                  ariaLabel={`Favorite ${item.id}`}
+                  iconName="star"
+                  pressedIconName="star-filled"
+                  variant="icon"
+                />
+              ) : undefined,
+              content: item.content,
+              secondaryContent: item.secondaryContent ? (
+                <Box color="text-status-inactive">{item.secondaryContent}</Box>
+              ) : undefined,
+              announcementLabel: item.announcementLabel,
+              actions: item.actions,
+            })}
+            getItemId={item => item.id}
+            getItemChildren={item => item.children}
+            i18nStrings={{
+              expandButtonLabel: () => 'Expand item',
+              collapseButtonLabel: () => 'Collapse item',
+            }}
+            connectorLines="vertical"
+          />
+        </Container>
+      </Grid>
+
+      <Button>Focus here</Button>
+
+      <Grid gridDefinition={[{ colspan: { m: 7, xs: 12 } }]}>
+        <Container header={<h2>Focusable non-CDS elements inside tree-item</h2>}>
```

Comment:
nit: I would rather say "registered" vs "non-registered" elements. There are cloudscape components like inputs or segmented controls that are not registered to single tab stop nav